lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -149,13 +176,24 @@ class RenderBlockingResources extends Audit {
    * @param {Node} fcpGraph
    * @param {Set<string>} deferredIds
    * @param {Map<string, number>} wastedCssBytesByUrl
+   * @param {LH.Artifacts.DetectedStack[]} Stacks
    * @return {number}
    */
-  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl) {
-    const originalEstimate = simulator.simulate(fcpGraph).timeInMs;
+  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl, Stacks) {
+    const originalEstimate = simulator.simulate(fcpGraph, {label: 'rbr-estimate'}).timeInMs;
+    const originalNodeTimings = Simulator.ALL_NODE_TIMINGS.get('rbr-estimate');
+
+    let stackSpecificEstimate = originalEstimate;
 
     let totalChildNetworkBytes = 0;
     const minimalFCPGraph = /** @type {NetworkNode} */ (fcpGraph.cloneWithRelationships(node => {
+      if (originalNodeTimings) {
+        const nodeTiming = originalNodeTimings.get(node);
+        if (nodeTiming) {
+          const stackSpecificTiming = computeStackSpecificTiming(node, nodeTiming, Stacks);
+          stackSpecificEstimate -= nodeTiming.duration - stackSpecificTiming.duration;
```

Comment:
I'm trying to wrap my head around how this works, so let me see if I follow what you're trying to do here.

- we lookup the node timing in the "before" version of the graph
- compute its stack-specific timing
- take the difference from the non-stack-specific timing and subtract it from the overall "before" estimate
- this lowers the effective "before" estimate to better reflect the fact that we might be overestimating it

right?

This is on the right track but I think it will run into issues with parallel requests. 

Consider the page dependency graph

```
|-index.html 1000ms
    |- style-1.css 2000ms
    |- style-2.css 2000ms
    |- style-3.css 2000ms
```

In this case we'll subtract 3 full seconds off the before time which was only 3s to begin with. A fully correct solution in this style would require threading the duration changes all the way down through core lantern, but that's a bit more involved than I think we need here :)

Instead, you could adjust all of the `nodeTimings` coming from `simulate` and recompute `const originalEstimate` as the `Math.max(...Array.from(adjustedNodeTimings.map(timing => timing.endTime)))`, you will also have to adjust the endTime of any dependent requests of the stylesheet (see line 137 for an example of how to find dependent nodes)